### DISCUSSION TITLE
Feature Request: Allow Custom Axios Instance in "axios" Mode as parameter

### DISCUSSION THREAD

**Discussion by FR073N** [orval-labs/orval#1617] [Ideas]
Currently, the "axios" mode in Orval generates code that uses either : 
- default Axios instance.
- or the one given by the mutator function.

What can be good, is to propose a mode to let the client give his **own axiosInstance**. This should be only possible with the `axios` **mode**.

#### Before : 

```ts
export const getPublicApi = () => {
  const postAuthLogin = <TData = AxiosResponse<LoginTokens>>(
    postApiPublicAuthLoginBody: PostApiPublicAuthLoginBody,
    options?: AxiosRequestConfig
  ): Promise<TData> => {
    return axios.post(`/api/public/auth/login`, postApiPublicAuthLoginBody, options);
  };
```

#### After : 

```ts
export const getPublicApi = (axiosInstance: AxiosInstance) => {
  const postAuthLogin = <TData = AxiosResponse<LoginTokens>>(
    postApiPublicAuthLoginBody: PostApiPublicAuthLoginBody,
    options?: AxiosRequestConfig
  ): Promise<TData> => {
    return axiosInstance.post(`/api/public/auth/login`, postApiPublicAuthLoginBody, options);
  };
```

This would allow for more flexibility and control over the network requests, especially in cases where multiple instances with different configurations are needed.

This feature would simplify the code and make it more versatile, as it would allow for the management of multiple cases with different Axios instances.

By using the provided axiosInstance instead of the default axios, the generated code becomes more flexible and adaptable to various project requirements.

This can be achieved as something as simple as : 

```js
      ...
      client: 'axios',
      axiosInstanceAsOptions: true, // an option like that
      mode: 'tags',
      target: './api/index.ts',
      workspace: './src/'
```

**Comment by samouds** [comment]
Agree.

It will be great 👍

**Comment by FR073N** [comment]
Any news on this ?
